packages/flame_rive/lib/src/rive_component.dart

```diff
@@ -0,0 +1,177 @@
+import 'dart:async';
+import 'dart:math';
+import 'dart:ui' as ui;
+
+import 'package:flame/components.dart';
+import 'package:flutter/rendering.dart';
+import 'package:rive/rive.dart';
+
+// ignore_for_file: implementation_imports
+import 'package:rive/src/rive_core/math/aabb.dart';
+import 'package:rive/src/rive_core/math/mat2d.dart';
+import 'package:rive/src/rive_core/math/vec2d.dart';
+
+class RiveComponent extends PositionComponent {
+  final Artboard artboard;
+  final RiveArtboardRenderer _renderer;
+
+  RiveComponent({
+    required this.artboard,
+    bool antialiasing = true,
+    bool useArtboardSize = true,
+    BoxFit fit = BoxFit.contain,
+    Alignment alignment = Alignment.center,
+
+    // position component arguments
+    Vector2? position,
+    Vector2? size,
+    Vector2? scale,
+    double angle = 0.0,
+    Anchor anchor = Anchor.topLeft,
```

Comment:
Don't these two already have these defaults directly in `PositionComponent`?

Reply:
Dart doesn't forward defaults. Super won't accept nullable arguments.

Reply:
Right, makes sense. Hmm, it doesn't seem like we have these arguments in other classes that extend `PositionComponent`, I think either we should add them to all (not in this PR), or we should remove them here so that we have some consistency.

Reply:
That woudl break soam example I did, soo...

Reply:
The examples would just have to use the `..` notation instead, depending on what we decide here.

Reply:
Since we decided to add there arguments to all `PositionComponent`s these can stay and I will sort out the defaults later.